File: Python/ClassAndInheritance.py
class FirstClass:
    def __init__(self):
        print("Constructor of Base Class")
    def methodOfBaseClass(self):
        print("Method of BaseClass")

class SecondClass(FirstClass):
    def __init__(self):
        super().__init__();#Use super method to call constructor of base class
        print("Constructor of Derived Class")
    def methodOfDerivedClass(self):
        print("Method of Derived Class")
        
#driver code

sc=SecondClass();
sc.methodOfBaseClass();
sc.methodOfDerivedClass()

File: JavaScript/ClassAndInheritance.js
class FirstClass {
  constructor() {
    console.log("This is constructor");
  }
  methodOfBaseClass() {
    console.log("BaseClass");
  }
}
class SecondClass extends FirstClass {
  constructor() {
    super(); // use super method to call constructor of base class
    console.log("Derived Constructor");
  }
  methodOfDerivedClass() {
    console.log("Derived Class");
  }
}

function Implementation() {
  var sc = new SecondClass();
  sc.methodOfDerivedClass();
  sc.methodOfBaseClass();
}
Implementation();

//Must call constructor of base class
